### PR TITLE
Add hard area limit for OTF

### PR DESCRIPTION
**Pull request by jterry64**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [X] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
No limits for OTF analysis query.

Issue Number: GTC-1214

## What is the new behavior?

- When getting geometry, return a shared geostore object containing geojson, area, bbox, and ID.
- Use that shared geostore object to impose hard limit of 1 billion hectares for raster analysis queries. Return 413 status code (payload too big) if it's over the limit.
- Remove some commented out imports from queries.py to remove pre-commit issues.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: app/utils/rw_api.py [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
 from ..errors import BadResponseError, InvalidResponseError, UnauthorizedError
 from ..models.pydantic.authentication import SignUp
-from ..models.pydantic.geostore import Geometry
+from ..models.pydantic.geostore import Geometry, GeostoreCommon
 from ..settings.globals import RW_API_URL
 
 
 @alru_cache(maxsize=128)
-async def get_geostore_geometry(geostore_id: UUID) -> Geometry:
+async def get_geostore_geometry(geostore_id: UUID) -> GeostoreCommon:
```
Comment (thomas-maschler) [inline app/utils/rw_api.py:16]: rename function

File: app/routes/datasets/queries.py [hunk 2]
```diff
@@ -531,19 +533,26 @@ async def _query_raster(
...
-    geometry: Geometry,
+    geostore: GeostoreCommon,
     format: QueryFormat = QueryFormat.json,
     delimiter: Delimiters = Delimiters.comma,
 ) -> Dict[str, Any]:
+    if geostore.area__ha > GEOSTORE_SIZE_LIMIT_OTF:
+        raise HTTPException(
+            status_code=413,
+            detail=f"Geostore area exceeds limit of {GEOSTORE_SIZE_LIMIT_OTF} ha for raster analysis.",
+        )
```
Comment (thomas-maschler) [inline app/routes/datasets/queries.py:544]: add test

File: app/routes/datasets/queries.py [hunk 3]
```diff
@@ -531,19 +533,26 @@ async def _query_raster(
...
     asset: AssetORM,
     sql: str,
-    geometry: Geometry,
+    geostore: GeostoreCommon,
     format: QueryFormat = QueryFormat.json,
     delimiter: Delimiters = Delimiters.comma,
 ) -> Dict[str, Any]:
+    if geostore.area__ha > GEOSTORE_SIZE_LIMIT_OTF:
+        raise HTTPException(
+            status_code=413,
```
Comment (thomas-maschler) [inline app/routes/datasets/queries.py:542]: i would make this either a `400` or `422` error

**Review by thomas-maschler — CHANGES_REQUESTED** [review]

File: app/utils/geostore.py [hunk 4]
```diff
@@ -8,29 +7,36 @@
...
+            geojson=geostore.gfw_geojson,
+            area__ha=geostore.gfw_area__ha,
+            bbox=geostore.gfw_bbox,
+        )
     except (KeyError, RecordNotFoundError) as ex:
         logger.exception(ex)
         raise BadResponseError("Cannot fetch geostore geometry")
 
-    if geometry is None:
+    if geostore.gfw_geojson is None:
```
Comment (dmannarino) [inline app/utils/geostore.py:30]: Hmm, is this right? Shouldn't the attribute name be just "geojson", so `if geostore.geojson is None:`?

File: app/utils/rw_api.py (hunk 1 above)
Comment (jterry64) [reply app/utils/rw_api.py]: done

File: app/utils/geostore.py (hunk 4 above)
Comment (jterry64) [reply app/utils/geostore.py:30]: Oh, I guess either way is correct, because I have copy of both the GFW geostore and geostore_common

**Review by dmannarino — APPROVED** [review]

**Pull request merged**